### PR TITLE
[Feature] Implementing erase function

### PR DESCRIPTION
**Pull request by cchobi**
# 배경
신규요구 사항인 erase 기능 구현 및 추가 필요

# 변경 내용
- HostIf에 Erase 명령어 validity check 추가 
- Erase command 추가
- NAND모듈에 erase 기능 추가

# 테스트 완료
```bash
[==========] Running 36 tests from 5 test suites.
[----------] Global test environment set-up.
[----------] 11 tests from FileManagerTestFixture
[ RUN      ] FileManagerTestFixture.FileManager
[       OK ] FileManagerTestFixture.FileManager (0 ms)
[ RUN      ] FileManagerTestFixture.FileManagerRead
[       OK ] FileManagerTestFixture.FileManagerRead (1 ms)
[ RUN      ] FileManagerTestFixture.FileManagerWrite
[       OK ] FileManagerTestFixture.FileManagerWrite (1 ms)
[ RUN      ] FileManagerTestFixture.FileManagerReadFileTest
[       OK ] FileManagerTestFixture.FileManagerReadFileTest (2 ms)
[ RUN      ] FileManagerTestFixture.FileManagerAllZeroReadFileTest
[       OK ] FileManagerTestFixture.FileManagerAllZeroReadFileTest (7 ms)
[ RUN      ] FileManagerTestFixture.FileManagerWriteFileTest
[       OK ] FileManagerTestFixture.FileManagerWriteFileTest (10 ms)
[ RUN      ] FileManagerTestFixture.FileManagerWriteInvalidOffset
[       OK ] FileManagerTestFixture.FileManagerWriteInvalidOffset (1 ms)
[ RUN      ] FileManagerTestFixture.FileManagerWriteNegativeOffset
[       OK ] FileManagerTestFixture.FileManagerWriteNegativeOffset (1 ms)
[ RUN      ] FileManagerTestFixture.FileManagerReadNegativeOffset
[       OK ] FileManagerTestFixture.FileManagerReadNegativeOffset (1 ms)
[ RUN      ] FileManagerTestFixture.FileManagerReadInvalidOffset
[       OK ] FileManagerTestFixture.FileManagerReadInvalidOffset (2 ms)
[ RUN      ] FileManagerTestFixture.FileNotOpen
[       OK ] FileManagerTestFixture.FileNotOpen (0 ms)
[----------] 11 tests from FileManagerTestFixture (34 ms total)

[----------] 8 tests from SSDIntegrationTest
[ RUN      ] SSDIntegrationTest.ReadTest
[       OK ] SSDIntegrationTest.ReadTest (6 ms)
[ RUN      ] SSDIntegrationTest.WriteReadTest
[       OK ] SSDIntegrationTest.WriteReadTest (4 ms)
[ RUN      ] SSDIntegrationTest.InvalidCommandTest
[       OK ] SSDIntegrationTest.InvalidCommandTest (0 ms)
[ RUN      ] SSDIntegrationTest.InvalidReadArguments
[       OK ] SSDIntegrationTest.InvalidReadArguments (0 ms)
[ RUN      ] SSDIntegrationTest.InvalidWriteArguments
[       OK ] SSDIntegrationTest.InvalidWriteArguments (1 ms)
[ RUN      ] SSDIntegrationTest.InvalidWriteLinenumber
Invalid LBA !!!
[       OK ] SSDIntegrationTest.InvalidWriteLinenumber (1 ms)
[ RUN      ] SSDIntegrationTest.InvalidWriteDataLength
Invalid data length !!!
[       OK ] SSDIntegrationTest.InvalidWriteDataLength (1 ms)
[ RUN      ] SSDIntegrationTest.InvalidWriteDataNoPrefix_0x
Invalid Data: Data should start with 0x !!! AB
[       OK ] SSDIntegrationTest.InvalidWriteDataNoPrefix_0x (1 ms)
[----------] 8 tests from SSDIntegrationTest (23 ms total)

[----------] 1 test from SSDTest
[ RUN      ] SSDTest.NANDInterface
[       OK ] SSDTest.NANDInterface (0 ms)
[----------] 1 test from SSDTest (0 ms total)

[----------] 12 tests from HostIntfTestFixture
[ RUN      ] HostIntfTestFixture.CheckingInvalidArgumentNum
[       OK ] HostIntfTestFixture.CheckingInvalidArgumentNum (0 ms)
[ RUN      ] HostIntfTestFixture.CheckingValidWriteCommands
[       OK ] HostIntfTestFixture.CheckingValidWriteCommands (0 ms)
[ RUN      ] HostIntfTestFixture.CheckingValidReadCommands
[       OK ] HostIntfTestFixture.CheckingValidReadCommands (0 ms)
[ RUN      ] HostIntfTestFixture.CheckingInvalidLBA
LBA is out of range !!!
[       OK ] HostIntfTestFixture.CheckingInvalidLBA (0 ms)
[ RUN      ] HostIntfTestFixture.CheckingInvalidData
Invalid Data !!!
[       OK ] HostIntfTestFixture.CheckingInvalidData (0 ms)
[ RUN      ] HostIntfTestFixture.StartWriteCmd
[       OK ] HostIntfTestFixture.StartWriteCmd (0 ms)
[ RUN      ] HostIntfTestFixture.StartReadCmd
[       OK ] HostIntfTestFixture.StartReadCmd (0 ms)
[ RUN      ] HostIntfTestFixture.dataStringCheck
Invalid Data !!!
[       OK ] HostIntfTestFixture.dataStringCheck (0 ms)
[ RUN      ] HostIntfTestFixture.WrongCommandNameCheck
Invalid Data !!!
[       OK ] HostIntfTestFixture.WrongCommandNameCheck (0 ms)
[ RUN      ] HostIntfTestFixture.CheckingValidEraseCommand
[       OK ] HostIntfTestFixture.CheckingValidEraseCommand (0 ms)
[ RUN      ] HostIntfTestFixture.CheckingInvalidEraseSize
Invalid erase size !!!
[       OK ] HostIntfTestFixture.CheckingInvalidEraseSize (0 ms)
[ RUN      ] HostIntfTestFixture.CheckingInvalidEraseArguementNum
[       OK ] HostIntfTestFixture.CheckingInvalidEraseArguementNum (0 ms)
[----------] 12 tests from HostIntfTestFixture (10 ms total)

[----------] 4 tests from NANDTest
[ RUN      ] NANDTest.NANDWriteRead
[       OK ] NANDTest.NANDWriteRead (1 ms)
[ RUN      ] NANDTest.NANDError
[       OK ] NANDTest.NANDError (5 ms)
[ RUN      ] NANDTest.NANDEraseTooLargeSize
[       OK ] NANDTest.NANDEraseTooLargeSize (10 ms)
[ RUN      ] NANDTest.NANDEraseSmallLeftSize
[       OK ] NANDTest.NANDEraseSmallLeftSize (2 ms)
[----------] 4 tests from NANDTest (23 ms total)

[----------] Global test environment tear-down
[==========] 36 tests from 5 test suites ran. (97 ms total)
[  PASSED  ] 36 tests.
```

### CODE REVIEW

File: gTest/TestNand.txt
Comment (6akjunyoung) [inline gTest/TestNand.txt:1]: 해당 file 변경점은 트래킹 안되도록 .gitignore에 추가해보긴 했는데 잘 반영이 안되었나보군요ㅠㅠ

File: gTest/test.cpp [hunk 1]
```diff
@@ -139,4 +173,22 @@ TEST(NANDTest, NANDError) {
...
+	nand.write(13, "0x1298DEAD");
+	nand.erase(0, eraseSize);
+}
+
+TEST(NANDTest, NANDEraseSmallLeftSize) {
+	NAND nand{ "TestNand.txt", "TestResult.txt" };
+	string eraseSize = "5";
+
+	nand.write(99, "0x1298DEAD");
+	nand.erase(98, eraseSize);
```
Comment (6akjunyoung) [inline gTest/test.cpp:193]: error case에 대해 TC 잘 구현해주셨군요! 고생하셨습니다~

**Review by 6akjunyoung — APPROVED** [review]

File: gTest/test.cpp (hunk 1 above)
Comment (jun6ak) [inline gTest/test.cpp:193]: boundary case까지 TC 잘 구현해주셨군요! 고생하셨습니다~

**Review by jun6ak — APPROVED** [review]

File: gTest/TestNand.txt
Comment (jun6ak) [inline gTest/TestNand.txt:1]: 해당 file 변경점은 트래킹 안되도록 .gitignore에 추가해보긴 했는데 잘 반영이 안되었나보네요ㅠㅠ

**Review by jun6ak — APPROVED** [review]
